web/packages/teleport/src/Player/Player.tsx

```diff
@@ -42,16 +42,40 @@ export function Player() {
   const { sid, clusterId } = useParams<UrlPlayerParams>();
   const { search } = useLocation();
 
+  useEffect(() => {
+    document.title = `Play ${sid} • ${clusterId}`;
+  }, [sid, clusterId]);
+
   const recordingType = getUrlParameter(
     'recordingType',
     search
   ) as RecordingType;
-  const durationMs = Number(getUrlParameter('durationMs', search));
+
+  // In order to render the progress bar, we need to know the length of the session.
+  // All in-product links to the session player should include the session duration in the URL.
+  // Some users manually build the URL based on the session ID and don't specify the session duration.
+  // For those cases, we make a separate API call to get the duration.
+  const [fetchDurationAttempt, fetchDuration] = useAsync(
+    useCallback(async () => {
+      const response = await fetchSessionDuration(clusterId, sid);
+      return response;
+    }, [clusterId, sid])
+  );
```

Comment:
We don't do anything with the awaited response, we can return the promise directly:
```suggestion
  const [fetchDurationAttempt, fetchDuration] = useAsync(
    useCallback(() => fetchSessionDuration(clusterId, sid), [clusterId, sid])
  );
```